resources/tests/EasyCrypt_GenDoc_Tutorial.ec

```diff
@@ -0,0 +1,311 @@
+  order listed above for the given `<item-name>` and links to the first match.
+  E.g., instead of `[go to type t above](>Ty|t)`, you can use `[go to type t
+  above](>|t)` (proof: [go to type t above](>|t)). However, you still need
+  to explicitly specify the `Op` kind for operator `t` below, because the
+  documentation tool checks the `Type` kind before the `Operator` kind, which
+  already results in a match.
+&*)
+op f : u -> v.
+
+(*& This is a documented operator &*)
+op t : v -> w.
+
+(*&
+  The generated documentation file contains a section for each item kind. Within each
+  section, items are displayed in the order they appear in the source.
+&*)
+axiom ax : true.
+
+(*&
+  If there are no items of a certain kind,
+  the generated documentation file does not contain a section for that kind.
+  For example, the generated documentation file for this theory does not
+  contain a section for module types.
+
+  The navigation bar on the left-hand side of the generated documentation file
+  shows only the sections that are present and provides links to them
+  for convenience.
+&*)
+lemma lem : true.
+proof. by trivial. qed.
+
+(*&
+  Currently, individual procedures inside of modules (and module types) cannot
+  be documented using documentation comments. For the time being, a
+  (unsatisfactory) workaround is to use regular comments within the module (or
+  module type), which appear in the source code for the module (or module type)
+  in the generated documentation file.
+&*)
+module M = {
+  (*
+    This regular comment appears in the source code
+    for this module in the generated documentation file.
+  *)
+  proc p() : int = {
+    return 1;
+  }
+}.
+
+(*&
+  Theories are special as "documentable" items. They appear as documented items
+  in the generated documentation file for their parent theory __and__
+  receive their own documentation file, in turn documenting all their
+  "documentable" items. This file has a subheading indicating the subtheory and
+  links to entry of the (sub)theory in the parent theory's documentation file.
+
+  The file name for a (sub)theory follows this pattern: `Y!Z`, where `Y`, and
+  `Z` represent the parent theory and (sub)theory, respectively. (This works
+  recursively: `Y` may itself be a (sub)theory of another theory `X`, in which
+  case the file name becomes `X!Y!Z`.)
+
+  The "introductory text" for the (sub)theory, which you would usually put in
+  file documentation comments, is drawn from the regular documentation comments
+  in the parent theory. In the parent theory's documentation file, the
+  (sub)theory's name links to the corresponding (sub)theory documentation file.
+
+  No source code is shown for subtheories in the parent theory's documentation
+  file.
+&*)
+theory T.
+
+(*&
+  This item is documented in the documentation file corresponding to
+  (sub)theory `T'.
+&*)
+type s.
+
+(*&
+  Linking to items is done from the perspective of the outermost theory (`Top`),
```

Comment:
Note here the difference between `Top` and `Self`. We should document it if we're going to be anal about it in documenting the documentation tool, but it is an important distinction to make somewhere.

Reply:
I meant this as just "Check that this is `Top` and not `Self`.", but my mind went "hold up, nobody knows the difference."

In this particular case, and since the tool runs on a single source file, I don't think there is one.

Reply:
Right, I am pretty sure I didn't see `Self` pop up anywhere while testing, but `Top` did each time. Then again, I don't know the difference myself. One of the next features on the list would be to implement multi-file documentation generation, so might as well make sure we get this right now. Can you summarize the difference of each?